### ISSUE TITLE
[LessSpecificImplementedReturnType] phpDoc not inherited when adding a more specified native type

### ISSUE THREAD

**Issue by iluuu1994** [vimeo/psalm#1593]
Not sure if this is intentional

```php
<?php

class Foo
{
  /**
   * @return int[]
   */
  public function doFoo()
  {
    return [1, 2, 3];
  }
}

class Bar extends Foo
{
  public function doFoo(): array
  {
    return parent::doFoo();
  }
}
```

> ERROR: LessSpecificImplementedReturnType - 16:28 - The return type 'array<array-key, int>' for Foo::doFoo is more specific than the implemented return type for Bar::dofoo 'array<array-key, mixed>'

https://psalm.dev/r/483659eb7f

`Bar::doFoo` added a more specific native return type but IMO the phpDoc type should still be inherited. Otherwise the phpDoc types just have to be copied manually.

**Comment by muglug** [comment]
Oh thanks for this - this was reported by @jwage last week, but from a more convoluted scenario. This is a great test case.

**Comment by iluuu1994** [comment]
Cool, thank you 🙂
